syft/formats/common/spdxhelpers/to_format_model.go

```diff
@@ -514,6 +514,21 @@ func toPackageOriginator(p pkg.Package) *spdx.Originator {
 	}
 }
 
+func toPackageSupplier(p pkg.Package) *spdx.Supplier {
+	// this uses the Originator function for now until
+	// a better distinction can be made for supplier
+	kind, supplier := Originator(p)
+	if kind == "" || supplier == "" {
+		return &spdx.Supplier{
+			Supplier: NOASSERTION,
+		}
+	}
```

Comment:
I think supplier is optional in SPDX, should we return `nil`?

Reply:
I wrote in the PR description for why this case exists:
```
The SPDX document creator has attempted to but cannot reach a reasonable objective determination
```

I don't think it's a question of if it's required or optional by SPDX, but more so did we attempt to provide an answer for this field. 

I think including `NOASSERTION` for cases we failed to identify a supplier is a better indicator on improving the SBOMs quality rather than removing the field entirely. Returning nil here might communicate to some users that we didn't even attempt to identify the supplier of a given source.

Reply:
After discussion this seems like the right thing to do for now 👍